src/pool/srv_pool.c

```diff
@@ -6653,6 +6691,31 @@ pool_svc_update_map_internal(struct pool_svc *svc, unsigned int opc,
 		goto out_map;
 	}
 
+	/* For SWIM exclude, don't change pool map if the pw_rf is broken or is going to be broken,
+	 * with CRIT log message to ask administrator to bring back the engine.
+	 */
+	if (src == MUS_SWIM && opc == MAP_EXCLUDE) {
+		d_rank_t	rank;
+		int		failed_cnt;
+
+		rc = pool_map_update_failed_cnt(map);
+		if (rc) {
+			DL_ERROR(rc, "pool_map_update_failed_cnt failed.");
+			goto out_map;
+		}
+
+		D_ASSERT(tgt_addrs->pta_number == 1);
```

Comment:
So swim can guarantee to exclude ranks one by one? For example, in Aurora configuration, if some server node crashed, then two ranks will be excluded in order or together?

Reply:
it is in the path of handle_event() -> pool_svc_exclude_rank() -> pool_svc_update_map() -> pool_svc_update_map_internal().
In pool_svc_exclude_rank() it always set list.pta_number = 1 to call pool_svc_update_map now.
So I think here it is safe to assert it for current code.